src/AutoRest.CSharp.V3/Output/Models/Serialization/Json/JsonPropertySerialization.cs

```diff
@@ -1,19 +1,21 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using AutoRest.CSharp.V3.Generation.Writers;
+
 namespace AutoRest.CSharp.V3.Output.Models.Serialization.Json
 {
     internal class JsonPropertySerialization
     {
-        public JsonPropertySerialization(string name, string? memberName, JsonSerialization valueSerialization)
+        public JsonPropertySerialization(string name, CodeWriterDelegate? memberName, JsonSerialization valueSerialization)
         {
             Name = name;
             MemberName = memberName;
             ValueSerialization = valueSerialization;
         }
 
         public string Name { get; }
-        public string? MemberName { get; }
+        public CodeWriterDelegate? MemberName { get; }
```

Comment:
This change was to facilitate constants, since constant properties were written the same for serialization as normal properties. However, for flattening, we don't have a property name, so we make this delegate write the actual constant value directly; which we already had support for in both the property writer (assignment of the property) and then as a constant in the request parts.

Reply:
I'm not a fan because it kind of break the abstraction, you can't deserialize into a constant.

Can we handle constants same way we did with object properties here where we still generate a parameter with a default value until we figure out exactly what semantics are expected.

Reply:
The parameter flattening is only used for serialization, not deserialization, so there isn't a use case where MemberName would represent a constant in deserialization, given the way flattened parameter JsonPropertySerialization is created. 

Our overall serialization story is fractured and needs to be normalized. Right now, there is a mix between de/serialization access and de/serialization construction that is non-uniform. I still like the idea of writing 1 serializer method and 1 deserializer method per nested (model) object, even if the object isn't used. It makes our generator code much cleaner since all de/serialization is always written to 1 place in the generated output. The more different places you try to reuse the same generated output logic, the more complex the generator's code becomes.

I believe this change shows that any data that is to be written and has the potential to be swapped out should be a CodeWriterDelegate, as it allows the information to not be pre-created as a string; it has a dynamic aspect to it. It also allows it to always  work with our type system (formattable string writers) or any other system we tie into code writing.

I don't like the idea of adding a parameter for which the user could provide and break the operation. However, the more reasonable option would be to have a place to inject writing a variable to represent the constant, and use that as the MemberName. But, that requires a lot more 'spaghetti logic' to an already complex hierarchy of calls.

Reply:
Reverted this back to string, since we're generating a constant parameter instead now.